systems/plants/@RigidBodyManipulator/doKinematics.m

```diff
@@ -52,7 +52,7 @@
     valuecheck(b_compute_second_derivatives, []);
```

Comment:
I know these are not from you, but valuechecks here seem superfluous (and potentially expensive), especially given the logic you've now added below.